cdap-common/bin/functions.sh

```diff
@@ -493,11 +491,34 @@ cdap_set_spark() {
         return 1
       fi
       SPARK_HOME=$(echo -e "${SPARK_VAR_OUT}" | grep ^SPARK_HOME= | cut -d= -f2)
-      export SPARK_HOME
-      return 0
     fi
-    return 1
   fi
+
+  export SPARK_HOME
+
+  # Find environment variables setup via spark-env.sh
+  SPARK_ENV_SH=${SPARK_HOME}/conf/spark-env.sh
```

Comment:
We don't always run spark-shell, right? I can take the SPARK_CONF_DIR as well, which basically is the logic used by spark